java-checks/src/main/java/org/sonar/java/checks/security/SecureCookieCheck.java

```diff
@@ -157,20 +160,32 @@ private void addToUnsecuredCookies(AssignmentExpressionTree assignment) {
   private void checkSecureCall(MethodInvocationTree mit) {
     if (isSetSecureCall(mit) && mit.methodSelect().is(Tree.Kind.MEMBER_SELECT)) {
       ExpressionTree methodObject = ((MemberSelectExpressionTree) mit.methodSelect()).expression();
+      Boolean secureArgument = getBooleanValue(mit.arguments().get(0));
+      Boolean isFalse = secureArgument != null && !secureArgument;
       if (methodObject.is(Tree.Kind.IDENTIFIER)) {
         IdentifierTree identifierTree = (IdentifierTree) methodObject;
-        if (!LiteralUtils.isFalse(mit.arguments().get(0))) {
+        if (!isFalse) {
           unsecuredCookies.remove(identifierTree.symbol());
         } else if (identifierTree.symbol().owner().isMethodSymbol()) {
           unsecuredCookies.add((Symbol.VariableSymbol) identifierTree.symbol());
         }
-      } else if (LiteralUtils.isFalse(mit.arguments().get(0))) {
+      } else if (isFalse) {
         // builder method
         unsecuredSetters.add(mit);
       }
     }
   }
 
+  @Nullable
+  private static Boolean getBooleanValue(ExpressionTree expression) {
+    Boolean value = ConstantUtils.resolveAsBooleanConstant(expression);
+    if (value == null && expression.is(Tree.Kind.IDENTIFIER)) {
```

Comment:
expression.is(Tree.Kind.IDENTIFIER) never fails, you could add the following test: cookie.setSecure(!value);